### ISSUE TITLE
False positives in java.lang.correctness.hardcoded-conditional

### ISSUE THREAD

**Issue by mschwager** [returntocorp/sgrep-rules#616]
Consider some findings from the Eclipse project:

```
/tmp/eclipse.platform.ui/bundles/org.eclipse.e4.ui.css.core/src/org/eclipse/e4/ui/css/core/serializers/CSSSerializer.java
severity:error rule:java.lang.correctness.hardcoded-conditional: useless if statement, always the same behavior
157:		if (firstSelector == false) {
158:			writer.write("\n\n");

/tmp/eclipse.platform.ui/bundles/org.eclipse.e4.ui.css.swt/src/org/eclipse/e4/ui/css/swt/helpers/URI.java
severity:error rule:java.lang.correctness.hardcoded-conditional: useless if statement, always the same behavior
2858:		if (!isHierarchical() ||
2859:				!equals(scheme, prefix.scheme(), true) ||
2860:				!equals(authority, prefix.authority()) ||
2861:				!equals(device, prefix.device()) ||
2862:				hasAbsolutePath() != prefix.hasAbsolutePath())
2863:		{
2864:			return null;

/tmp/eclipse.platform.ui/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/fragment/impl/StringModelFragmentImpl.java
severity:error rule:java.lang.correctness.hardcoded-conditional: useless if statement, always the same behavior
382:				if (elements.isEmpty() == false) {
383:					ret.addAll(ModelUtils.merge(targetElement, feature, elements, getPositionInList()));
```

Using `if (!something) { ... }` may be preferable to `if (something == false) { ... }`, but it's still fine to write the latter version. There were a lot of these false positives on the Eclipse project, and I expect elsewhere too. Let's update this rule to get rid of these false positives.

**Comment by sfc-gh-jsalassi** [comment]
seeing this same issue on a large java project

**Comment by colleend** [comment]
seems like the false positives seen here don't flag anymore  with Grayson's fix: https://semgrep.dev/s/3rvo. I'll close this issue!
